invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/ItemDetails/state/actions.js

```diff
@@ -25,3 +37,36 @@ export const fetchItemDetails = itemPid => {
       });
   };
 };
+
+export const deleteItem = (itemPid, ownProps) => {
+  return async dispatch => {
+    dispatch({
+      type: DELETE_IS_LOADING,
+    });
+
+    await itemApi
+      .del(itemPid)
+      .then(response => {
+        dispatch({
+          type: DELETE_SUCCESS,
+          payload: { itemPid: itemPid },
+        });
+        setTimeout(() => {
+          dispatch(
+            sendSuccessNotification(
+              'Success!',
+              `The item ${itemPid} has been deleted.`
+            )
+          );
+          ownProps.history.push(BackOfficeURLS.itemsSearch);
```

Comment:
Ok I see why ownprops was needed here, but I addressed this issue in the past that's why there is this [history.js](https://github.com/inveniosoftware/invenio-app-ils/blob/master/invenio_app_ils/ui/src/invenio_app_ils/history.js). This prevents us by just using `withRouter` in order to redirect urls. See how it is used [here](https://github.com/inveniosoftware/invenio-app-ils/blob/master/invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/ItemDetails/components/ItemMetadata/components/CreateNewLoanModal/CreateNewLoanModal.js#L50). I think we need to refactor all the places that we do that as the actual purpose of `withRouter` is to expose the component in the router and be able to route to it.